### PR TITLE
add ocr e2e pgnet model

### PR DESCRIPTION
**Pull request by wxsIcey**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: mindocr/models/heads/e2e_head.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    """
+    PGNet Head module for text detection.
+    This module takes a single input feature map and applies convolutional operations
+    to generate the output predicts.
+    Args:
+       in_channels (int): The feature dimension of a single feature map generated by the neck (E2eFpn).
+    Returns:
+       Tensor: The output predict of f_score, f_boder, f_char, f_direction.
+    """
+    def __init__(self, in_channels: int, **kwargs):
```
Comment (can-gaa-hou) [inline mindocr/models/heads/e2e_head.py:17]: kwargs如果不用，可以去掉，减少调用栈开销

File: mindocr/models/necks/e2e_neck.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+class E2eFpn(nn.Cell):
+    """
+    PGNet Feature Pyramid Network (FPN) module for text detection.
+    Args:
+        in_channels (List[int]): The input channel dimensions for feature map.
+        out_channels (int): The output channel size.
+    Returns:
+        Tensor: The output feature map of shape [batch_size, out_channels, H, W].
+    """
+    def __init__(self, in_channels, **kwargs):
```
Comment (can-gaa-hou) [inline mindocr/models/necks/e2e_neck.py:15]: 同上，如果kwargs不用可以去掉

File: mindocr/models/backbones/e2e_resnet.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
+from mindspore import nn
+
+from ..utils import ConvNormLayer
+from ._registry import register_backbone, register_backbone_class
+from .mindcv_models.resnet import default_cfgs
+from .mindcv_models.utils import load_pretrained
+__all__ = ['E2ePgResNet', 'pgnet_backbone']
```
Comment (hongziqi) [inline mindocr/models/backbones/e2e_resnet.py:8]: 需要增加空行；建议本地下载 pre-commit之后，验证一遍再提交。
（pip install pre-commit; pre-commit run --all-files）

Comment (wxsIcey) [reply mindocr/models/backbones/e2e_resnet.py:8]: 嗯嗯，本地pre-commit有问题，已经解决ci问题了

**Review by can-gaa-hou — APPROVED** [review]

**Comment by can-gaa-hou** [comment]
lgtm

**Pull request merged**